docs/encyclopedia/application-message-passing.mdx

```diff
@@ -26,57 +28,85 @@ keywords:
 - temporal sdk message passing
 ---
 
+import PrettyImage from '@site/src/components/pretty-image/PrettyImage';
 import { RelatedReadContainer, RelatedReadItem } from '@site/src/components/related-read/RelatedRead';
 
-Temporal applications can receive and provide data through messages.
+Workflows can be thought of as stateful web services that can receive messages. The Workflow can have powerful message handlers akin to endpoints that can react to the incoming messages in combination with the current state of the Workflow.
 Temporal supports three types of messages: Signals, Queries, and Updates.
+To define these, consider these messages from the perspective of the client making the request:
 
-## Signals {#signals}
+- Queries are read requests. They can read the current state of the Workflow but cannot block in doing so.
+- Signals are write requests. They cause changes in the running Workflow, but you cannot await any response or error.
+- Updates are interactive write requests, wherein the sender of the Update tracks completion and waits for a response or an error.
 
-**What is a Signal in the context of Temporal?**
+## How to choose between Signals, Updates, and Queries? {#choosing-messages}
 
-A Signal is an asynchronous request to a [Workflow Execution](/workflows#workflow-execution).
+### For write requests
 
-<RelatedReadContainer>
-    <RelatedReadItem path="/cli/workflow#signal" text="Send a Signal using the Temporal CLI" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/go/message-passing#signals" text="Send and handle Signals with the Go SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/java/message-passing" text="Send and handle Signals with the Java SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/php/message-passing#signals" text="Send and handle Signals with the PHP SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/python/message-passing#signals" text="Send and handle Signals with the Python SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/typescript/message-passing#signals" text="Send and handle Signals with the TypeScript SDK" archetype="feature-guide" />
-    <RelatedReadItem path="/develop/dotnet/message-passing#define-a-signal" text="Send and handle Signals with the .NET SDK" archetype="feature-guide" />
-</RelatedReadContainer>
+From the above descriptions, it may sound like Updates can do all the things that Queries and Signals can do. But, unlike Signals, Updates must be synchronous, that is, they must wait for the worker running the Workflow to acknowledge the request.
+
+Use Signals instead of Updates when:
+
+- You want to send a fully asynchronous message.
+- Your client does not want to rely on the worker being available.
+- Your Workflow’s client cares more about the latency of sending the message than it does of the end-to-end operation.
```

Comment:
There's a typo (`s/does of/does about/`) but, more importantly, I'm finding that I have to re-read this one multiple times. At least part of my brain is saying: "But a client who chooses signal over update _does_ care about the end-to-end latency: it thinks that it is too slow to be able to wait for it". I think perhaps something more like the following is clearer:

```suggestion
- Your Workflow’s client wants the overall latency to be low, and is prepared to give up the ability to track the operation in return for low latency.
```

Reply:
Your rephrasing isn't quite what I was emphasizing, as I addressed trackability in other points.  However, I made some minor readability tweaks.